code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -17,58 +17,566 @@
+	</br>The pylon will fire weak beams that are harmless to the cult. In addition it can be upgraded even more by shooting it with a laser, which will give it a limited number of extra-power shots."
+
+	icon_state = "pylonbase"
 	var/isbroken = 0
 	light_range = 5
 	light_color = "#3e0000"
-	var/obj/item/wepon = null
+	var/pylonmode = 0
+	//Pylon has three states:
+	//0 = Idle, the pylon is an inert crystal
+	//1 = Awaiting sacrifice. The pylon is actively tracking a creature to be sacrificed to it
+	//2 = Turret. The pylon has been empowered by a sacrifice and is now a turret permanantly
+
+	var/damagetaken = 0
+	var/empowered = 0 //Number of empowered, higher-damage shots remaining
+	var/processing = 0
+	var/mob/living/sacrifice //Holds a reference to a mob that is a pending sacrifice
+	var/mob/living/sacrificer //A reference to the last mob that attempted to sacrifice something. So we can message them
+	//Sacrifier is also used in target handling. the pylon will not bite the hand that feeds it. Noncultist colleagues are fair game though
+
+	var/next_shot = 0 //Absolute world time when we're allowed to fire again
+	var/shot_delay = 5 //Minimum delay between shots, in deciseconds
+	var/mob/living/target
+
+	var/notarget //Number of times handle_firing has been called without finding anything to shoot at
+	//This is used for switching to lower-intensity processing
+
+	var/datum/language/cultcommon/lang
+	//An instance of the cult language datum. Used to scramble speech when speaking to noncultists
+
+	var/turf/last_target_loc
+
+	var/process_interval = 1
+	var/ticks
+	anchored = 0
+
+
+//Just a subtype that starts off in turret mode. For adminbus and debugging. Maybe a future wizard spell
+//Spawn them next to ERPers
+/obj/structure/cult/pylon/turret
+	pylonmode = 2
+
+/obj/structure/cult/pylon/turret/New()
+	..()
+	start_process()
+
+
+//Another subtype which starts with infinite empower shots. For empowered adminbus
+/obj/structure/cult/pylon/turret/empowered
+	empowered = 99999999
+
+
+/obj/structure/cult/pylon/New()
+	..()
+	lang = new /datum/language/cultcommon()
+	update_icon()
+
+/obj/structure/cult/pylon/examine(var/mob/user)
+	..()
+	if (damagetaken)
+		switch (damagetaken)
+			if (1 to 8)
+				user << "It has very faint hairline fractures."
+			if (8 to 20)
+				user << "It has several cracks across its surface."
+			if (20 to 30)
+				user << "It is chipped and deeply cracked, it may shatter with much more pressure."
+			if (30 to INFINITY)
+				user << "It is almost cleaved in two, the pylon looks like it will fall to shards under its own weight."
+
+
+/obj/structure/cult/pylon/Move()
+	..()
+	last_target_loc = null
+
+/obj/structure/cult/pylon/proc/start_process()
+	process_interval = 1
+	if (!processing)
+		processing_objects += src
+		processing = 1
+
+
+//If the pylon goes a long time without shooting anything, it will consider slowing down processing
+/obj/structure/cult/pylon/proc/reconsider_interval()

```

Comment:
View with a small radius is also cheap. Your method will probably not target people riding mules (which `mobs_in_view` compensates for).

This seems like a lot of complexity to 'optimize' an object that will have very few instances on the map.

Turrets' target acquisition is more complex than this, and yet does not show significant CPU usage in the profiler.